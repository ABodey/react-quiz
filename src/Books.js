import React, { PureComponent } from 'react';

class Books extends PureComponent {
state = {
    books: []
}

    onAddBook = (event) => {
        event.preventDefault();

        const author = event.target.author.value
        const title = event.target.title.value

        this.setState({
            books: [
                ...this.state.books,
                {
                    _id: v1(),
                    author,
                    title,
                },
            ]
        })
        event.target.reset();
    }

    render() {
        return (
            <div >
                {books.length ? (
                    <ul> {books.map((book => {
                        <Book { ...book} key={book._id}
                        />
                    }))}
                    </ul>    
                ) : (
                    <p> None found </p>
                )}
                <AddBook onAddBook={this.onAddBook} />
            </div>
        );
    }
}

export default Books;
