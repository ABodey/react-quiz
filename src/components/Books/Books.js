import React from 'react'
import Book from '../Book/Book'
import AddBook from '../AddBook/AddBook'


const Books = () => (
  <div>
  <h1> Books </h1>
  <Book />
  <AddBook />
    </div>
)

export default Books

