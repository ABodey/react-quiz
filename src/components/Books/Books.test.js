import React from 'react'
import { shallow } from 'enzyme'
import Books from './Books'

describe('<Books />', () => {
    it ('should match snapshots', () => {
      const wrapper = shallow (<Books />)

      expect(wrapper).toMatchSnapshot()
    })
})

