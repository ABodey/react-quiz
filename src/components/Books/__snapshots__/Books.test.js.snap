// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Books /> should match snapshots 1`] = `
<div>
  <h1>
     Books 
  </h1>
  <Book />
  <AddBook />
</div>
`;
